File: app/src/main/java/fr/paug/androidmakers/util/UIUtils.java
package fr.paug.androidmakers.util;

import android.content.res.ColorStateList;
import android.os.Build;
import android.support.annotation.RequiresApi;
import android.util.Property;
import android.view.View;

public class UIUtils {

    public static final Property<View, Integer> BACKGROUND_TINT
            = new Property<View, Integer>(Integer.class, "backgroundTint") {

        @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
        @Override
        public void set(View view, Integer color) {
            view.setBackgroundTintList(ColorStateList.valueOf(color));
        }

        @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
        @Override
        public Integer get(View view) {
            return view.getBackgroundTintList().getDefaultColor();
        }
    };

}

File: app/src/main/java/fr/paug/androidmakers/util/UIUtils.kt
package fr.paug.androidmakers.util

import android.content.res.ColorStateList
import android.util.Property
import android.view.View

object UIUtils {
  val BACKGROUND_TINT: Property<View, Int> =
      object : Property<View, Int>(Int::class.java, "backgroundTint") {
        override fun set(view: View, color: Int) {
          view.backgroundTintList = ColorStateList.valueOf(color)
        }

        override fun get(view: View): Int {
          return view.backgroundTintList!!.defaultColor
        }
      }
}